database.go

```diff
@@ -1,37 +1,81 @@
 package main
 
 import (
-	"database/sql"
+	"errors"
+	"fmt"
+	"strings"
 )
 
-var db *sql.DB
+var (
+	ErrDBConnStrParseNoSep = errors.New("Missing ':' in connection string")
+	ErrDBConnStrParseNotImplemented = func(dbName string) error {
```

Comment:
Ok, I've made all errors to be returned by functions. I've added an errors.go file to make a clean interface for this process.
Strings like "err.conn.parse.sqlite.file.missing" will document what error we're returning (they are essentially self documenting" and extra arguments can be added for the error functions that take them.